### ISSUE TITLE
Rebuild crate when artefact contents change

### ISSUE THREAD

**Issue by taminomara** [cowprotocol/ethcontract-rs#540]
Changes in artefact `.json` files should trigger a rebuild.

**Comment by nlordell** [comment]
I thought this was already the case (because they are `include!`-ed) but it may have gotten broken in the past.

**Comment by taminomara** [comment]
I'm not sure if it works with `build.rs`, but with macros it certainly isn't the case.

**Comment by taminomara** [comment]
So it looks like we can't add dependency on actual JSON files unless we `include!` them 😕

**Comment by nlordell** [comment]
If we just `include!` them, but don't do anything with the file, does the optimizer remove the data from the output binary? If so, we can
```rust
let _ = include_bytes!("source/path.json");
```

**Comment by taminomara** [comment]
Yes, the optimizer will remove the data. However, the compiler will still read that data and include it into AST (at least HIR, not sure about MIR). This will potentially slow down compilation. Not sure if it's a real concern though.

**Comment by nlordell** [comment]
> Yes, the optimizer will remove the data. However, the compiler will still read that data and include it into AST (at least HIR, not sure about MIR). This will potentially slow down compilation. Not sure if it's a real concern though.

Yeah, its a downside for sure - but personally I view not recompiling as a bigger issue than the slower compile time. Is there a macro to include the cargo fingerprints without reading it?

**Comment by taminomara** [comment]
> Is there a macro to include the cargo fingerprints without reading it?

I'm not sure, probably no.
